Cython/Compiler/Nodes.py

```diff
@@ -2883,7 +2883,12 @@ def analyse_declarations(self, env):
             self.is_staticmethod = False
 
         if self.name == '__new__' and env.is_py_class_scope:
-            self.is_staticmethod = 1
+            self.is_staticmethod = True
+        if not self.is_classmethod and self.name in ('__class_getitem__', '__init_subclass__') and env.is_py_class_scope:
+            from .ExprNodes import NameNode
+            self.decorators = self.decorators or []
+            self.decorators.append(DecoratorNode(self.pos, decorator=NameNode(self.pos, name='classmethod')))
```

Comment:
Need to double check whether this should be added to the front or back of decorators

Reply:
It'll be the outermost decorator, so probably the front I think. (Although at the minute it probably won't work if there are any other decorators present anyway - I have a PR in to fix this but it's big and some of the changes are unpopular)